### PR TITLE
Validate that this repository does not contain offensive language

### PR DESCRIPTION
**Pull request by dominikholler**
Add a check to the sannity test that fails if there is a usage
of offensive language in one of the files.

The script forbit the use of the words `master`, `slave`,
`blacklist` and `whitelist`

Sinse `master` is very common word as a name of default
branches or as a node name in clusters, the script allows
the usage of it if this word is part of a URL or a path.
The script does not check several directories that may
contain third party usage of the word `master`:
`vendor/` `deploy/` and `cluster`.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ endif
 all: manager
 
 # Run tests
-unittest: generate fmt vet manifests
+unittest: generate fmt vet manifests validate-no-offensive-lang
```
Comment (kwiesmueller) [inline Makefile:43]: Should we really still keep it in the unittests then?

Comment (kwiesmueller) [reply Makefile:43]: Ah, nevermind I thought you'd take the GitHub Action here as well (I'd prefer it).

**Comment by dominikholler** [comment]
> Ah, nevermind I thought you'd take the GitHub Action here as well (I'd prefer it).

I changed to github action.

**Comment by kwiesmueller** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**